File: minegame-server/src/main/java/top/nipuru/minegame/server/game/item/ItemData.java
package top.nipuru.minegame.server.game.item;

import lombok.Getter;
import lombok.experimental.Accessors;
import top.nipuru.minegame.server.game.player.Data;
import top.nipuru.minegame.server.game.player.Table;
import top.nipuru.minegame.server.game.player.Unique;

@Getter
@Accessors(fluent = true)
@Table(name = "tb_player")
public class ItemData implements Data {
    /** 物品类型 */
    @Unique int itemType;

    /** 物品id */
    @Unique int itemId;

    /** 数量 */
    long amount;
}

File: minegame-game/src/main/kotlin/top/nipuru/minegame/game/gameplay/item/ItemData.kt
package top.nipuru.minegame.game.gameplay.item

import top.nipuru.minegame.game.gameplay.player.Data
import top.nipuru.minegame.game.gameplay.player.Table
import top.nipuru.minegame.game.gameplay.player.Unique

@Table(name = "tb_player")
class ItemData : Data {
    /** 物品类型  */
    @Unique
    var itemType: Int = 0

    /** 物品id  */
    @Unique
    var itemId: Int = 0

    /** 数量  */
    var amount: Long = 0
}
